docs/development.md

```diff
@@ -1,54 +1,113 @@
 # Oak Development
 
+- [Meta-Advice](#meta-advice)
+- [Prerequisites](#prerequisites)
+- [Run Example Application](#run-example-application)
+  - [Build Application](#build-application)
+  - [Build Runtime Server](#build-runtime-server)
+  - [Run Runtime Server](#run-runtime-server)
+  - [Run Example Client](#run-example-client)
+
+## Meta-Advice
+
+For any open-source project, the best way of figuring out what prerequisites and
+dependendencies are required is to look at the projects continuous integration
+(CI) scripts. A project's top-level [Travis](https://travis-ci.org/),
+[CircleCI](https://circleci.com/),
+[Cloudbuild](https://cloud.google.com/cloud-build) or
+[Dockerfile](https://docs.docker.com/) will give definitive information about
+what's needed, and at what version.
+
+For Oak, the CI system is currently Cloudbuild using a Docker image, so the
+top-level [`cloudbuild.yaml`](/cloudbuild.yaml) and [`Dockerfile`](Dockerfile)
+hold the project's configuration, dependencies and build scripts.
+
 ## Prerequisites
 
-We use Docker to install the base dependencies that need to exist at the system
-level, e.g. the [Rust compiler](https://www.rust-lang.org/tools/install) and the
-[Bazel build system](https://bazel.build); these steps are detailed in
-[`Dockerfile`](/Dockerfile). See
-https://docs.docker.com/engine/reference/builder/ for more information.
-
-Inside Docker, we use Bazel to version, install and build any non-Rust
-dependencies, including our own code. Dependencies are listed in
-[`WORKSPACE`](/WORKSPACE). See
-https://docs.bazel.build/versions/master/external.html for more information.
-
-To set up your development environment, you need the following applications. For
-the accurate versions required for a successful build please consult the current
-[`Dockerfile`](/Dockerfile).
-
-- **Docker**: follow
-  [Docker install instructions](https://docs.docker.com/install)
-- **Bazel**: follow
-  [Bazel install instructions](https://docs.bazel.build/versions/master/install.html)
-- **Rust**:
-  - Follow install instructions from https://rustup.rs/:
+The project's [`Dockerfile`](/Dockerfile) holds the definitive set of
+prerequisites for building and running Oak. The steps involved are generally
+clear from the contents, but the
+[Docker documentation](https://docs.docker.com/engine/reference/builder/) gives
+details for any steps that aren't obvious.
+
+The key prerequisites are:
+
+- **Docker**: If you want to use Docker to ensure an exact match with the
+  project's official build environment, follow the
+  [Docker install instructions](https://docs.docker.com/install).
+
+  If you want to work on the project without using Docker, you should
+  synchronize your installed versions of all tools with those specified in the
+  [`Dockerfile`](/Dockerfile).
```

Comment:
Why do we support both ways? It seems like saying "just use docker" might be pretty straightforward? 

Personally I'd be inclined to place the docker instructions inside of a "Get Started" section, and the non-docker approach inside of an "Advanced" section.

Reply:
Maybe make this a "choose your own adventure" doc (with corresponding headings): either install Docker, or here are a few hints to what you may want to install on your host (but still the Dockerfile is the ultimate source of truth). You may want to do both if you end up developing Oak, since the iteration cycle is faster if the tools are installed natively, and your editor can integrate with them directly in that case.

Reply:
Yeah I think that would be awesome. 

Docker as a way to get started quickly, works out of the box,
Native gets you extra speed & integration, at the cost of potentially needing to figure out things on your own.